itest-flake: bitcoind commitment deadline

```
=== RUN   TestLightningNetworkDaemon/59-of-82/bitcoind/commitment_deadline
    lnd_channel_force_close.go:135: 
        	Error Trace:	lnd_channel_force_close.go:135
        	            				lnd_channel_force_close.go:194
        	            				test_harness.go:112
        	            				lnd_test.go:11702
        	Error:      	Received unexpected error:
        	            	error while waiting for broadcast tx: wanted 9c0592bda73dbc6e83a17f7c0ccc2fbe13887406aaa43e0847e7e9e84fa4b84b, found 0 txs in mempool: []
        	Test:       	TestLightningNetworkDaemon/59-of-82/bitcoind/commitment_deadline
        	Messages:   	unable to force close channel
=== CONT  TestLightningNetworkDaemon
    lnd_test.go:11710: Failure time: 2021-07-05 07:09:55.560
--- FAIL: TestLightningNetworkDaemon (1039.98s)
    --- PASS: TestLightningNetworkDaemon/41-of-82/bitcoind/switch_offline_delivery_persistence (44.93s)
    --- PASS: TestLightningNetworkDaemon/42-of-82/bitcoind/switch_offline_delivery_outgoing_offline (30.42s)
    --- PASS: TestLightningNetworkDaemon/43-of-82/bitcoind/revoked_uncooperative_close_retribution (16.38s)
    --- PASS: TestLightningNetworkDaemon/44-of-82/bitcoind/failing_link (24.86s)
    --- PASS: TestLightningNetworkDaemon/45-of-82/bitcoind/garbage_collect_link_nodes (73.67s)
    --- PASS: TestLightningNetworkDaemon/46-of-82/bitcoind/abandonchannel (20.67s)
    --- PASS: TestLightningNetworkDaemon/47-of-82/bitcoind/revoked_uncooperative_close_retribution_zero_value_remote_output (18.43s)
    --- PASS: TestLightningNetworkDaemon/48-of-82/bitcoind/revoked_uncooperative_close_retribution_remote_hodl (53.78s)
    --- PASS: TestLightningNetworkDaemon/49-of-82/bitcoind/revoked_uncooperative_close_retribution_altruist_watchtower (59.07s)
        --- PASS: TestLightningNetworkDaemon/49-of-82/bitcoind/revoked_uncooperative_close_retribution_altruist_watchtower/anchors (27.01s)
        --- PASS: TestLightningNetworkDaemon/49-of-82/bitcoind/revoked_uncooperative_close_retribution_altruist_watchtower/legacy (28.83s)
    --- PASS: TestLightningNetworkDaemon/50-of-82/bitcoind/data_loss_protection (73.17s)
    --- PASS: TestLightningNetworkDaemon/51-of-82/bitcoind/query_routes (22.89s)
    --- PASS: TestLightningNetworkDaemon/52-of-82/bitcoind/route_fee_cutoff (35.43s)
    --- PASS: TestLightningNetworkDaemon/53-of-82/bitcoind/send_update_disable_channel (66.89s)
    --- PASS: TestLightningNetworkDaemon/54-of-82/bitcoind/streaming_channel_backup_update (34.38s)
    --- PASS: TestLightningNetworkDaemon/55-of-82/bitcoind/export_channel_backup (14.14s)
    --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore (389.77s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_from_RPC_backup (39.40s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_from_backup_file (42.02s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_during_creation (39.44s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_during_unlock (47.02s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_from_backup_file_twice (36.86s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_unconfirmed_channel_file (40.08s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_unconfirmed_channel_RPC (53.36s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_from_backup_file_anchors (45.64s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/old_revocation_producer_format (42.76s)
    --- PASS: TestLightningNetworkDaemon/57-of-82/bitcoind/hold_invoice_sender_persistence (9.42s)
    --- PASS: TestLightningNetworkDaemon/58-of-82/bitcoind/hold_invoice_force_close (14.57s)
    --- FAIL: TestLightningNetworkDaemon/59-of-82/bitcoind/commitment_deadline (34.63s)
```

Neutrino variant: 
```
    lnd_channel_force_close.go:157: 
        	Error Trace:	lnd_channel_force_close.go:157
        	            				lnd_channel_force_close.go:194
        	            				test_harness.go:112
        	            				lnd_test.go:11702
        	Error:      	Received unexpected error:
        	            	wanted 2, found 1 txs in mempool: [6652e7ce90372ae1f32ac54aff201084b43ecf672c4c18414866d9ada68a78b9]
        	Test:       	TestLightningNetworkDaemon/59-of-82/neutrino/commitment_deadline
        	Messages:   	failed to find commitment tx in mempool
=== CONT  TestLightningNetworkDaemon
    lnd_test.go:11710: Failure time: 2021-07-05 07:09:40.331
--- FAIL: TestLightningNetworkDaemon (1058.47s)
    --- PASS: TestLightningNetworkDaemon/41-of-82/neutrino/switch_offline_delivery_persistence (34.91s)
    --- PASS: TestLightningNetworkDaemon/42-of-82/neutrino/switch_offline_delivery_outgoing_offline (29.45s)
    --- PASS: TestLightningNetworkDaemon/43-of-82/neutrino/revoked_uncooperative_close_retribution (17.85s)
    --- PASS: TestLightningNetworkDaemon/44-of-82/neutrino/failing_link (18.51s)
    --- PASS: TestLightningNetworkDaemon/45-of-82/neutrino/garbage_collect_link_nodes (78.21s)
    --- PASS: TestLightningNetworkDaemon/46-of-82/neutrino/abandonchannel (16.99s)
    --- PASS: TestLightningNetworkDaemon/47-of-82/neutrino/revoked_uncooperative_close_retribution_zero_value_remote_output (19.76s)
    --- PASS: TestLightningNetworkDaemon/48-of-82/neutrino/revoked_uncooperative_close_retribution_remote_hodl (50.74s)
    --- PASS: TestLightningNetworkDaemon/49-of-82/neutrino/revoked_uncooperative_close_retribution_altruist_watchtower (56.06s)
        --- PASS: TestLightningNetworkDaemon/49-of-82/neutrino/revoked_uncooperative_close_retribution_altruist_watchtower/anchors (26.34s)
        --- PASS: TestLightningNetworkDaemon/49-of-82/neutrino/revoked_uncooperative_close_retribution_altruist_watchtower/legacy (26.37s)
    --- PASS: TestLightningNetworkDaemon/50-of-82/neutrino/data_loss_protection (66.24s)
    --- PASS: TestLightningNetworkDaemon/51-of-82/neutrino/query_routes (29.82s)
    --- PASS: TestLightningNetworkDaemon/52-of-82/neutrino/route_fee_cutoff (50.04s)
    --- PASS: TestLightningNetworkDaemon/53-of-82/neutrino/send_update_disable_channel (50.60s)
    --- PASS: TestLightningNetworkDaemon/54-of-82/neutrino/streaming_channel_backup_update (33.66s)
    --- PASS: TestLightningNetworkDaemon/55-of-82/neutrino/export_channel_backup (25.72s)
    --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore (374.88s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_from_RPC_backup (42.89s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_from_backup_file (37.32s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_during_creation (38.16s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_during_unlock (40.79s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_from_backup_file_twice (38.39s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_unconfirmed_channel_file (40.62s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_unconfirmed_channel_RPC (38.56s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_from_backup_file_anchors (47.56s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/old_revocation_producer_format (42.31s)
    --- PASS: TestLightningNetworkDaemon/57-of-82/neutrino/hold_invoice_sender_persistence (10.21s)
    --- PASS: TestLightningNetworkDaemon/58-of-82/neutrino/hold_invoice_force_close (10.63s)
    --- FAIL: TestLightningNetworkDaemon/59-of-82/neutrino/commitment_deadline (80.90s)
```

> ```
> === RUN   TestLightningNetworkDaemon/59-of-82/bitcoind/commitment_deadline
>     lnd_channel_force_close.go:135: 
>         	Error Trace:	lnd_channel_force_close.go:135
>         	            				lnd_channel_force_close.go:194
>         	            				test_harness.go:112
>         	            				lnd_test.go:11702
>         	Error:      	Received unexpected error:
>         	            	error while waiting for broadcast tx: wanted 9c0592bda73dbc6e83a17f7c0ccc2fbe13887406aaa43e0847e7e9e84fa4b84b, found 0 txs in mempool: []
>         	Test:       	TestLightningNetworkDaemon/59-of-82/bitcoind/commitment_deadline
>         	Messages:   	unable to force close channel
> === CONT  TestLightningNetworkDaemon
>     lnd_test.go:11710: Failure time: 2021-07-05 07:09:55.560
> --- FAIL: TestLightningNetworkDaemon (1039.98s)
>     --- PASS: TestLightningNetworkDaemon/41-of-82/bitcoind/switch_offline_delivery_persistence (44.93s)
>     --- PASS: TestLightningNetworkDaemon/42-of-82/bitcoind/switch_offline_delivery_outgoing_offline (30.42s)
>     --- PASS: TestLightningNetworkDaemon/43-of-82/bitcoind/revoked_uncooperative_close_retribution (16.38s)
>     --- PASS: TestLightningNetworkDaemon/44-of-82/bitcoind/failing_link (24.86s)
>     --- PASS: TestLightningNetworkDaemon/45-of-82/bitcoind/garbage_collect_link_nodes (73.67s)
>     --- PASS: TestLightningNetworkDaemon/46-of-82/bitcoind/abandonchannel (20.67s)
>     --- PASS: TestLightningNetworkDaemon/47-of-82/bitcoind/revoked_uncooperative_close_retribution_zero_value_remote_output (18.43s)
>     --- PASS: TestLightningNetworkDaemon/48-of-82/bitcoind/revoked_uncooperative_close_retribution_remote_hodl (53.78s)
>     --- PASS: TestLightningNetworkDaemon/49-of-82/bitcoind/revoked_uncooperative_close_retribution_altruist_watchtower (59.07s)
>         --- PASS: TestLightningNetworkDaemon/49-of-82/bitcoind/revoked_uncooperative_close_retribution_altruist_watchtower/anchors (27.01s)
>         --- PASS: TestLightningNetworkDaemon/49-of-82/bitcoind/revoked_uncooperative_close_retribution_altruist_watchtower/legacy (28.83s)
>     --- PASS: TestLightningNetworkDaemon/50-of-82/bitcoind/data_loss_protection (73.17s)
>     --- PASS: TestLightningNetworkDaemon/51-of-82/bitcoind/query_routes (22.89s)
>     --- PASS: TestLightningNetworkDaemon/52-of-82/bitcoind/route_fee_cutoff (35.43s)
>     --- PASS: TestLightningNetworkDaemon/53-of-82/bitcoind/send_update_disable_channel (66.89s)
>     --- PASS: TestLightningNetworkDaemon/54-of-82/bitcoind/streaming_channel_backup_update (34.38s)
>     --- PASS: TestLightningNetworkDaemon/55-of-82/bitcoind/export_channel_backup (14.14s)
>     --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore (389.77s)
>         --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_from_RPC_backup (39.40s)
>         --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_from_backup_file (42.02s)
>         --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_during_creation (39.44s)
>         --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_during_unlock (47.02s)
>         --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_from_backup_file_twice (36.86s)
>         --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_unconfirmed_channel_file (40.08s)
>         --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_unconfirmed_channel_RPC (53.36s)
>         --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/restore_from_backup_file_anchors (45.64s)
>         --- PASS: TestLightningNetworkDaemon/56-of-82/bitcoind/channel_backup_restore/old_revocation_producer_format (42.76s)
>     --- PASS: TestLightningNetworkDaemon/57-of-82/bitcoind/hold_invoice_sender_persistence (9.42s)
>     --- PASS: TestLightningNetworkDaemon/58-of-82/bitcoind/hold_invoice_force_close (14.57s)
>     --- FAIL: TestLightningNetworkDaemon/59-of-82/bitcoind/commitment_deadline (34.63s)
> ```

Haven't seen this for `bitcoind` for a while.

--- a/contractcourt/channel_arbitrator.go
+++ b/contractcourt/channel_arbitrator.go
@@ -1100,7 +1100,7 @@ func (c *ChannelArbitrator) sweepAnchors(anchors *lnwallet.AnchorResolutions,
 	// sweepWithDeadline is a helper closure that takes an anchor
 	// resolution and sweeps it with its corresponding deadline.
 	sweepWithDeadline := func(anchor *lnwallet.AnchorResolution,
-		htlcs htlcSet) error {
+		htlcs htlcSet, anchorPath string) error {
 
 		// Find the deadline for this specific anchor.
 		deadline, err := c.findCommitmentDeadline(heightHint, htlcs)
@@ -1109,7 +1109,8 @@ func (c *ChannelArbitrator) sweepAnchors(anchors *lnwallet.AnchorResolutions,
 		}
 
 		log.Debugf("ChannelArbitrator(%v): pre-confirmation sweep of "+
-			"anchor of tx %v", c.cfg.ChanPoint, anchor.CommitAnchor)
+			"anchor of %s commit tx %v", c.cfg.ChanPoint,
+			anchorPath, anchor.CommitAnchor)
 
 		// Prepare anchor output for sweeping.
 		anchorInput := input.MakeBaseInput(
@@ -1153,13 +1154,15 @@ func (c *ChannelArbitrator) sweepAnchors(anchors *lnwallet.AnchorResolutions,
 	for htlcSet, htlcs := range c.activeHTLCs {
 		switch {
 		case htlcSet == LocalHtlcSet && anchors.Local != nil:
-			err := sweepWithDeadline(anchors.Local, htlcs)
+			err := sweepWithDeadline(anchors.Local, htlcs, "local")
 			if err != nil {
 				return err
 			}
 
 		case htlcSet == RemoteHtlcSet && anchors.Remote != nil:
-			err := sweepWithDeadline(anchors.Remote, htlcs)
+			err := sweepWithDeadline(
+				anchors.Remote, htlcs, "remote",
+			)
 			if err != nil {
 				return err
 			}
@@ -1167,7 +1170,9 @@ func (c *ChannelArbitrator) sweepAnchors(anchors *lnwallet.AnchorResolutions,
 		case htlcSet == RemotePendingHtlcSet &&
 			anchors.RemotePending != nil:
 
-			err := sweepWithDeadline(anchors.RemotePending, htlcs)
+			err := sweepWithDeadline(
+				anchors.RemotePending, htlcs, "remote pending",
+			)
 			if err != nil {
 				return err
 			}

--- a/docs/release-notes/release-notes-0.14.0.md
+++ b/docs/release-notes/release-notes-0.14.0.md
@@ -114,6 +114,8 @@ you.
 
 * [Fixed context timeout when closing channels in tests](https://github.com/lightningnetwork/lnd/pull/5616).
 
+* [Fixed transaction not found in mempool flake in commitment deadline itest](https://github.com/lightningnetwork/lnd/pull/5615).
+
 * [Fixed a missing import and git tag in the healthcheck package](https://github.com/lightningnetwork/lnd/pull/5582).
 
 * [Fixed a data race in payment unit test](https://github.com/lightningnetwork/lnd/pull/5573).

--- a/lntest/itest/assertions.go
+++ b/lntest/itest/assertions.go
@@ -1613,8 +1613,9 @@ func assertActiveHtlcs(nodes []*lntest.HarnessNode, payHashes ...[]byte) error {
 
 			// Channel should have exactly the payHashes active.
 			if len(payHashes) != len(htlcHashes) {
-				return fmt.Errorf("node %x had %v htlcs active, "+
-					"expected %v", node.PubKey[:],
+				return fmt.Errorf("node [%s:%x] had %v "+
+					"htlcs active, expected %v",
+					node.Cfg.Name, node.PubKey[:],
 					len(htlcHashes), len(payHashes))
 			}
 
@@ -1624,9 +1625,9 @@ func assertActiveHtlcs(nodes []*lntest.HarnessNode, payHashes ...[]byte) error {
 				if _, ok := htlcHashes[h]; ok {
 					continue
 				}
-				return fmt.Errorf("node %x didn't have the "+
-					"payHash %v active", node.PubKey[:],
-					h)
+				return fmt.Errorf("node [%s:%x] didn't have: "+
+					"the payHash %v active", node.Cfg.Name,
+					node.PubKey[:], h)
 			}
 		}
 	}

--- a/lntest/itest/lnd_channel_force_close.go
+++ b/lntest/itest/lnd_channel_force_close.go
@@ -81,6 +81,13 @@ func testCommitmentTransactionDeadline(net *lntest.NetworkHarness,
 
 		// Send some coins to the node.
 		net.SendCoins(ctx, t.t, btcutil.SatoshiPerBitcoin, node)
+
+		// For neutrino backend, we need one additional UTXO to create
+		// the sweeping tx for the remote anchor.
+		if net.BackendCfg.Name() == lntest.NeutrinoBackendName {
+			net.SendCoins(ctx, t.t, btcutil.SatoshiPerBitcoin, node)
+		}
+
 		return node
 	}
 

--- a/lntest/itest/lnd_channel_force_close.go
+++ b/lntest/itest/lnd_channel_force_close.go
@@ -151,8 +151,7 @@ func testCommitmentTransactionDeadline(net *lntest.NetworkHarness,
 			t.t, checkNumWaitingCloseChannels(pendingChanResp, 1),
 		)
 
-		// We should see only one sweep transaction because the anchor
-		// sweep is skipped.
+		// Check our sweep transactions can be found in mempool.
 		sweepTxns, err := getNTxsFromMempool(
 			net.Miner.Client,
 			expectedSweepTxNum, minerMempoolTimeout,

